### DISCUSSION TITLE
Breaking change with new Prisma schema, how to migrate?

### DISCUSSION THREAD

**Discussion by baptisteArno** [nextauthjs/next-auth#5846] [Help]
Hi,

I can't upgrade to the latest version of `next-auth` because of the typing change. I have my own custom Prisma adapter. It now looks like email isn't nullable anymore?

But, how am I supposed to migrate this? I have tons of users who logged in using the Facebook provider, and it didn't save the email.

Related to https://github.com/nextauthjs/next-auth/pull/4953

_Originally posted by @baptisteArno in https://github.com/nextauthjs/next-auth/discussions/5708_

**Comment by ThangHuuVu** [comment]
I don't think we introduce any breaking changes in #4953, it's only TS changes
What problem are you running into? Could you help describe it a bit more? 🙌

**Comment by baptisteArno** [comment]
In the prisma schema, email used to be nullable:

```
model User {
  email         String?   @unique
}
```

The PR makes the email field required.

But in my database, I have lots of users without an email address (those who logged in with Facebook for example).
I just can't pass the email from nullable to required like that ? So the type wouldn't reflect what's in my database. Am I missing something ?

**Comment by ThangHuuVu** [comment]
I see what you mean... 🤔 Did you try to use [Prisma Migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate)?

**Reply by baptisteArno** [reply]
I can only do a migration from an optional field to a required one if the table doesn't contain any NULL field. Which is currently not the case 🤔

My underlying question is, are we sure that now, the email isn't nullable anymore?

Because, like I said, with the Facebook login option, I still don't collect the email as far as I know?

**Reply by baptisteArno** [reply]
The doc hasn't been updated either: https://next-auth.js.org/adapters/prisma#create-the-prisma-schema
